incydr/cli/cmds/file_events.py

```diff
@@ -136,45 +149,75 @@ def search(
 
     query.page_size = 10000
 
+    # Checkpointing
+    if checkpoint_name:
+        cursor = _get_cursor_store(client.settings.api_client_id)
+        checkpoint = cursor.get(checkpoint_name)  # or ""
+        if checkpoint:  # if stored checkpoint, overwrite query
+            query = EventQuery.parse_raw(checkpoint)
+
+        def checkpoint_func(event_):
+            q = query.copy()
+            q.page_token = event_["event"]["id"]
+            cursor.replace(checkpoint_name, json.dumps(q.dict()))
```

Comment:
Stored checkpoints are json strings of the original query withe the `pageToken` key updated to contain the ID of the last returned event